Wikipedia/Code/WMFWelcomePageViewController.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+import UIKit
+
+enum WMFWelcomePageType {
+    case intro
+    case languages
+    case analytics
+}
+
+public protocol WMFWelcomeNavigationDelegate: class{
+    func showNextWelcomePage(sender: AnyObject)
+}
+
+class WMFWelcomePageViewController: UIPageViewController, UIPageViewControllerDataSource, WMFWelcomeNavigationDelegate {
+
+    var completionBlock: (() -> Void)?
+    private var indexForDotIndicator:Int = 0
+    
+    func showNextWelcomePage(sender: AnyObject){
+        let index = pageControllers.indexOf(sender as! UIViewController)
+        if index == pageControllers.count - 1 {
+            self.dismissViewControllerAnimated(true, completion:completionBlock)
+        }else{
+            self.view.userInteractionEnabled = false
+            let nextIndex = index! + 1
+            indexForDotIndicator = nextIndex
+            self.setViewControllers([pageControllers[nextIndex]], direction: .Forward, animated: true, completion: {(Bool) in
+                self.view.userInteractionEnabled = true
+            })
+        }
+    }
+
+    private func containerControllerForWelcomePageType(type: WMFWelcomePageType) -> WMFWelcomeContainerViewController {
+        let controller = WMFWelcomeContainerViewController.wmf_viewControllerWithIdentifier("WMFWelcomeContainerViewController", fromStoryboardNamed: "WMFWelcome")
```

Comment:
Same thing here… wouldn't expect to need to pass a stringified version of the class name to the method which is a category on the class